subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/DefaultTaskOutputFilesRepository.java

```diff
@@ -19,6 +19,7 @@
 import org.gradle.cache.PersistentCache;
 import org.gradle.cache.PersistentIndexedCache;
 import org.gradle.cache.PersistentIndexedCacheParameters;
+import org.gradle.internal.snapshot.FileSystemSnapshotter;
```

Comment:
Is this right?

Reply:
Well spotted! Now that we can determine the roots from the output file fingerprints, the whole declared output file data is not necessary anymore. Moreover, `FileSystemSnapshotter.exists` can go away and we never hit the file system when updating the generated output files generation! Deleting all that code felt so good...